entities/Client.php

```diff
@@ -170,17 +170,22 @@ public function update(){
 	}
 
 	public function delete(){
-		$stmt=readOne($this->id);
-		// we need to parse in the orgId from clients.php
-		if($stmt->rowCount()==1){
-			// $sql = "DELETE FROM client_links WHERE client_id=:id AND link_id=:org_id; DELETE FROM client_needs WHERE client_id=:id AND requesting_organization_id=:org_id;";
-			$sql = "DELETE FROM clients WHERE id=:id"; // do NOT use this, it will delete clients across organisations
-			$stmt= $this->connection->prepare($sql);
-			return $stmt->execute(['id'=>$this->id]);
-		} else {
-			return false;
+        $stmt=$this->connection->prepare("select link_id,link_type from client_links where client_id=:id");
+		$stmt->execute(['id'=>$this->id]);
+		$active_org_id = $_SESSION['organization_id'];
+        if($stmt->rowCount()==1){
+			$row = $stmt->fetch();
+			if($active_org_id==$row['link_id']){
+				// TODO: add a database transaction				
+				$stmt= $this->connection->prepare("DELETE FROM clients WHERE id=:client_id; DELETE FROM client_needs WHERE client_id=:client_id; DELETE FROM need_requests WHERE client_need_id=:client_id; DELETE FROM client_links WHERE client_id=:client_id AND link_type='ORG'");
+				return $stmt->execute(['client_id'=>$this->id]);
```

Comment:
We seem to have lost the deletes of the other tables. Is there a reason for that?

Reply:
We integrated the deletes into one line, but from our testing, we saw that it will still delete from all four tables.

Reply:
got it. Thanks